pyinfra/connectors/util.py

```diff
@@ -13,17 +12,17 @@
 from pyinfra.api.util import memoize
 
 SUDO_ASKPASS_ENV_VAR = "PYINFRA_SUDO_PASSWORD"
-SUDO_ASKPASS_EXE_FILENAME = "pyinfra-sudo-askpass"
-
-
-def get_sudo_askpass_exe():
-    return StringIO(
-        """#!/bin/sh
-echo ${0}
+SUDO_ASKPASS_COMMAND = r"""
+temp=$(mktemp /tmp/pyinfra-sudo-askpass-XXXXXXXXXXXX)
+cat >"$temp"<<'__EOF__'
+#!/bin/sh
+printf '%s\n' "${0}"
```

Comment:
Why the switch to `printf` from `echo` here?

Reply:
It's a habit I got into because `printf '%s' ...` is safe, reliable, and portable. `echo` can cause trouble if, e.g., the password starts with a dash, then echo ends up taking that as an option. Not all shells have a built-in echo. More here:

https://unix.stackexchange.com/questions/65803/why-is-printf-better-than-echo

Reply:
Good habit to have! Definitely seems sensible to me, thank you!